### ISSUE TITLE
Add a block transform from navigation widget to navigation block

### ISSUE THREAD

**Issue by spacedmonkey** [WordPress/gutenberg#35780]
## What problem does this address?

Allow navigation widget to be transformed in navigation block in the block based widget editor. 

<img width="1062" alt="Screenshot 2021-10-19 at 21 52 33" src="https://user-images.githubusercontent.com/237508/137988837-374f4673-816f-4b6e-9b9d-36b48ba8a73a.png">

**Comment by noisysocks** [comment]
Makes sense. Noting that we can't enable the Navigation block until https://github.com/WordPress/gutenberg/issues/32952 is addressed, though, as Navigation stores its content into a Navigation entity.

**Comment by noisysocks** [comment]
Probably makes sense to tackle this and https://github.com/WordPress/gutenberg/issues/35781 at the same time.
